docs/en/pact-reference.md

```diff
@@ -599,6 +610,193 @@ Examples of valid keyset JSON productions:
+
+Additionally, interfaces my make use of module declarations, admitting use of the [use](pact-functions.html#use) keyword, allowing interfaces to import members of other modules. This allows interface signatures to be defined in terms of table types defined in an imported module.
+
+#### Example: Declaring and implementing an interface
+
+```lisp
+(interface my-interface
+    (defun hello-number:string (number:integer)
+        "Return the string \"Hello, $number!\" when given a string"
+        )
+
+    (defconst SOME_CONSTANT 3)
+)
+
+(module my-module (read-keyset 'my-keyset)
+
+    (implements my-interface)
+
+    (defun hello-number:string (number:integer)
+        (format "Hello, {}!" [number]))
+
+    (defun square-three ()
+        (* my-interface.SOME_CONSTANT my-interface.SOME_CONSTANT))
+)
+```
+
+### Declaring models in an interface
+
+[Formal verification](pact-properties.html) is implemented at multiple levels within an interface in order to provide an extra level of security. Models may be declared either within the body of the interface or at the function level in the same way that one would declare them in a module, with the exception that not all models are applicable to an interface. Indeed, since there is no abstract notion of tables for interfaces, abstract table invariants cannot be declared. However, if an interface imports table schema and types from a module via the [use](pact-functions.html#use) keyword, then the interface can define body and function models that apply directly to the concrete table type. Otherwise, all properties are candidates for declaration in an interface.
+
+When models are declared in an interface, they are appeneded to the list of models present in the implementing module at the level of declaration: body-level models are appended to body-level models, and function-level models are appended to function-level models. This allows users to extend the constraints of an interface with models applicable to specific business logic and implementation.
+
+Declaring models shares the same syntax with modules:
+
+#### Example: declaring models, tables, and importing modules in an interface
+
+```lisp
+(begin-tx)
+(env-data { "admin-keyset": ["product-dept" "dev-dept"] })
+
+(module acct-module ACCT-GOVERNANCE
+  @doc "account schema module"
+
+  (defcap ACCT-GOVERNANCE ()
+    true)
+
+  (defschema account
+    balance:integer
+    ks:keyset
+    active:bool
+  )
+)
+(commit-tx)
+(begin-tx)
+
+(interface coin-sig
+
+  "Coin Contract Abstract Interface Example"
+
+  (use acct-module)
+
+  (defun transfer:string (from:string to:string amount:integer)
+    @doc   "Transfer money between accounts"
+    @model [(property (row-enforced accounts "ks" from))
+            (property (> amount 0))
+            (property (= 0 (column-delta accounts "balance")))
```

Comment:
Referring to:
```lisp
(column-delta accounts "balance")
```
This is tricky because `accounts` table is created in another module that implements this interface and not in `acct-module`. Not sure if this _should_ be allowed, but maybe we can just add the following to `acct-module` to avoid any confusion:
```lisp
(deftable accounts:{account})
```

Reply:
Good call. Added.